torchmultimodal/models/video_vqvae.py

```diff
@@ -77,6 +90,76 @@ def video_vqvae(
     return VQVAE(encoder, decoder, num_embeddings, embedding_dim)
 
 
+def video_vqvae_mugen(
+    in_channel_dim: int = 3,
+    encoder_hidden_dim: int = 240,
+    encoder_kernel_size: int = 3,
+    n_res_layers: int = 4,
+    attn_hidden_dim: int = 240,
+    num_embeddings: int = 2048,
+    embedding_dim: int = 256,
+    decoder_hidden_dim: int = 240,
+    decoder_kernel_size: int = 3,
+    pretrained_model_key: Optional[str] = None,
+) -> VQVAE:
+    """Constructor for MUGEN's Video VQVAE. Expects input video data of shape {8,16,32}x256x256.
+    Trained for tokenization of video data and use in video-audio-text retrieval and generation tasks.
+    See Hayes et al. 2022 for more details: https://arxiv.org/pdf/2204.08058.pdf
+    Code ref:
+    https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/video_vqvae/vqvae.py
+    https://github.com/mugen-org/MUGEN_baseline/blob/main/generation/experiments/vqvae/VideoVQVAE_L32.sh
+
+    Args:
+        in_channel_dim (int, optional): Size of channel dim in input. Defaults to 3.
+        encoder_hidden_dim (int, optional): Size of channel dims in encoder conv layers. Defaults to 240.
+        encoder_kernel_size (int, optional): Kernel size for encoder. Defaults to 3.
+        n_res_layers (int, optional): Number of ``AttentionResidualBlocks`` to include in encoder and decoder. Defaults to 4.
+        attn_hidden_dim (int, optional): Size of hidden dim of ``AttentionResidualBlocks``. Defaults to 240.
+        num_embeddings (int, optional): Number of embedding vectors used in ``Codebook``. Defaults to 2048.
+        embedding_dim (int, optional): Dimensionality of embedding vectors in ``Codebook``. Defaults to 256.
+        decoder_hidden_dim (int, optional): Size of channel dims in decoder conv tranpose layers. Defaults to 240.
+        decoder_kernel_size (int, optional): Kernel size for decoder. Defaults to 3.
+        pretrained_model_key (str, optional): Load a specified MUGEN VQVAE checkpoint.
+
+    Returns:
+        VQVAE: constructed ``VQVAE`` model using ``VideoEncoder``, ``Codebook``, and ``VideoDecoder``
+    """
+    encoder_strides = ((2, 2, 2), (2, 2, 2), (1, 2, 2), (1, 2, 2), (1, 2, 2), (1, 1, 1))
+    decoder_strides = ((2, 2, 2), (2, 2, 2), (1, 2, 2), (1, 2, 2), (1, 2, 2))
+    encoder_n_layers = len(encoder_strides)
+    decoder_n_layers = len(decoder_strides)
+    encoder_in_channel_dims = (in_channel_dim,) + (encoder_hidden_dim,) * max(
+        encoder_n_layers - 1, 0
+    )
+    decoder_out_channel_dims = (decoder_hidden_dim,) * max(decoder_n_layers - 1, 0) + (
+        in_channel_dim,
+    )
+    encoder_kernel_sizes_fixed = _preprocess_int_conv_params(
```

Comment:
Although we chatted offline, just to flag here that the kernel size of the last layer is [`3` in MUGEN codebase](https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/video_vqvae/vqvae.py#L254).

I believe `_preprocess_int_conv_params` only treats uniform kernel sizes, right? Maybe for now we can just hard code the kernel sizes `((4, 4, 4), ...., (3, 3, 3))` (like you did for the strides) until we have a solution for it.

Reply:
actually, the kernel_size is set to 3 for both encoder and decoder in the [scripts](https://l.workplace.com/l.php?u=https%3A%2F%2Fgithub.com%2Fmugen-org%2FMUGEN_baseline%2Fblob%2Fmain%2Fgeneration%2Fexperiments%2Fvqvae%2FVideoVQVAE_L16.sh&h=AT2F0VAVmB3crWom9dUOnO2-HFn--2mq3wYuTbaTb0X5CcVUPMuMdzMoihmiMJCTbgOY0DnhCWSxZshQpDjxqdmRU4w6aebBatIJ9sV8JM8o85xhbV5DAwY1L6rY9LK8a4GJhOtExHxnhaWeWSPjE7ZwFI70TtC2LHvwTw). The paper also states that "The 3D VQ-VAE is similar to [69] except that we usea kernel size of 3"

Reply:
If this is something very specific to MUGEN, I will move `video_vqvae_mugen` to `examples/mugen/retrieval/` (I can do the move on my PR if you want).